### ISSUE TITLE
Add TextBundle support. 

### ISSUE THREAD

**Issue by JackalZhao** [billthefarmer/diary#138]
For a long time, inserting images to markdown document is always a problem: 

- Using image bed links is not stable. 
- Using local file path is messy for managing files, and also not stable. 

That's what really bothers me when using markdown to write. 

However, `Marked` developer and `Ulysses` team developed a excellent solution: the  **TextBundle**, which is based on **file system**, not *private package format*. May I explain it shortly, it is such kind of a structure: 

- A folder named `Document.textbundle` , includes: 
  - A folder named `assets` , which stores all the relevant media files (like images). 
  - A file named  `info.json` , which stores all the relevant metadata (tags, creating date, author, ......) 
  - A file named  `text.md` , which stores the main document, which uses relevant path to use media files in the  `assets` folder. 

It can achieve: including both images and md document within one package, no extra permission requires when previewing, and very convenient for editing directly. 

Every md document and it's relevant images are all stored in one TextBundle package, so that it's very convenient when exporting, copying one passage to another platform or device, every old and new Markdown editor can open, edit, and preview the md file,  so it is compatible with all Markdown editors. 

Softwares such as **Ulysses**, **Bear**, **Zettlr**  or even mind map tools like **Mindnode** are now supporting  **TextBundle** . 

If may support **TextBundle** , we can feel more safe and comfortable when writing. Because we can make sure all of our creation can be exported with all its images and other resources, and we can easily edit these resources using other professional tools. 

Here is the TextBundle [official site](http://textbundle.org/) and the [specifications](http://textbundle.org/spec/).

**Comment by billthefarmer** [comment]
It's a nice idea, but it is predicated on the macOS/iOS bundle concept, which is confined to that ecosystem. I could only find one library which is for macOS/iOS. Everything else &ndash; windows/linux/android, etc, uses archives of one sort or another &ndash; tar/zip/7z/rar etc.

**Comment by JackalZhao** [comment]
> It's a nice idea, but it is predicated on the macOS/iOS bundle concept, which is confined to that ecosystem. I could only find one library which is for macOS/iOS. Everything else – windows/linux/android, etc, uses archives of one sort or another – tar/zip/7z/rar etc.

Haha, we may have some misunderstanding I think. 😁

A TextBundle is of course not a private format package, the simplest form is a **folder** named **Document name.textbundle**, all the app need to do is just open and edit the text.md within this **Document name.textbundle** folder, treating the folder with names of **\*.textbundle** as a bundle.  We don't even need to use other packages like tar/zip/7z/rar, a **Bundle** is a **folder**. 

I know that one app called **Cola Notes** are currently supporting the textbundle. May I present one example of it's notes : (the bundle is the folder within the attatchment)

[25.textbundle.zip](https://github.com/billthefarmer/diary/files/4677137/25.textbundle.zip)

**Comment by JackalZhao** [comment]
So after a long discussion with the developer of [liandi](https://github.com/88250/liandi) yesterday, we got a simplified resolution. We decided to knock off the JSON magic. 

For each document, like a diary, here is it's structure: 

* a parent folder called `2020-05-27.textbundle` 
* a subfolder called `assets` 
* a document called `text.md` 

That's the simplest structure. The algorism is also simple: 

When opening a folder, check it's name, if the folder name adapts `*.textbundle` , then open the `text.md` inside it, else open the folder. And, for images, they will always be copied into the `assets` folder. 

Then we can get a document nice and teat. every document with its images and attachments are all in one folder, much easier to manage or copy the images in the future.  

I think this method can be applied to the Diary easily. When it does, then a diary path would be `Diary/2020/05/27.textbundle/text.md` instead of `Diary/2020/05/27.md` , then we can then get a much better structure. 

@billthefarmer

**Comment by billthefarmer** [comment]
I always use libraries to deal with external files in my apps. Java has built in XML, Properties and JSON libraries which I have used in other apps. I also use opencsv in one of my apps for reading CSV files. As I said before, the only textbundle library I can find is [TextBundle ](https://github.com/shinyfrog/TextBundle), written in Objective C for macOS and iOS and making use of the Foundation and CoreServices frameworks built into macOS and iOS.

As this, along with all my other apps, is free and open source, anyone with the necessary skills can fork it and add the functionality they want and possibly open a pull request. I forked this app originally.
